frontend/src/components/Displays/ConfirmScheduleDialogs/ConfirmScheduleDialog.tsx

```diff
@@ -23,17 +23,25 @@ export const ScheduleMissionWithConfirmDialogs = ({
     const { enabledRobots } = useRobotContext()
     const [robot, setRobot] = useState<Robot>()
 
-    const isBatteryInsufficient = (currentRobot: Robot) =>
-        currentRobot.batteryLevel &&
-        currentRobot.model.batteryWarningThreshold &&
-        currentRobot.batteryLevel < currentRobot.model.batteryWarningThreshold
+    const isBatteryInsufficient = (currentRobot: Robot) => {
+        const has_battery_value = currentRobot.batteryLevel !== null && currentRobot.batteryLevel !== undefined
```

Comment:
I would day the has_battery is kinda of a python thing and in typescript we would normally use hasBatteryValue? maybe change the variables names